src/types/types.ts

```diff
@@ -1,12 +1,35 @@
+export enum Region {
+  Africa = 'Africa',
+  Americas = 'Americas',
+  Asia = 'Asia',
+  Europe = 'Europe',
+  Oceania = 'Oceania',
+}
+
 export interface Country {
   name: {
     common: string;
   };
   population: number;
-  region: string;
+  region: Region;
   capital?: string;
   cca3: string;
   flags: {
     svg: string;
   };
 }
+
+export interface CountryDetails extends Country {
+  subregion: string;
+  borders?: string[];
+  currencies: {
+    [key: string]: {
+      name: string;
+      symbol: string;
+    };
+  };
+  languages: {
+    [key: string]: string;
```

Comment:
w takich przypadkach można też skorzystać z utility type Record, oznacza on to samo co te mapped types, czyli podajemy pierwszy parametr którym jest klucz i drugi którym jest wartość, czyli tutaj byłoby to: Record<string, string>